crates/compiler/parse/src/src64.rs

```diff
@@ -0,0 +1,437 @@
+// Loads Roc source files (from strings or from files) into a structure which is
+// guaranteed to have the following properties, all of which the SIMD parser requires:
+// - 16B alignment
+// - byte length is a multiple of 64
+// - if the source bytes were not a multiple of 64, the extra space is filled with trailing newlines
+//
+// (Trailing newlines are the filler of choice because they are irrelevant to the parser.)
+//
+// It does this as efficiently as possible by using branchless SIMD to fill padding bytes,
+// and reading the contents of the file directly into an arena in as few syscalls as possible.
+
+use bumpalo::{self, Bump};
+use core::{
+    alloc::Layout,
+    mem::{self, MaybeUninit},
+    ptr::{self, NonNull},
+    slice,
+};
+use std::{
+    fs::File,
+    io::{self, Read},
+    path::Path,
+};
+
+#[cfg(not(test))]
+/// We store both line and column numbers as u16s, so the largest possible file you could open
```

Comment:
Just curious, why do we store line and column number at all. I thought that most work around fast lexing and parsing would just calculate the values from the offset when needed. That way you can just store the offset and truly have a max length of 4GB.

Reply:
I believe the idea now is to not even store the offset. If there are errors to be reported, we can just reparse the file (that is still in memory) and then keep track of line and column for each node.

Reply:
I'm still not exactly sure how that should work (need to think through tradeoffs around the fact that canonicalization reorders nodes, for example) but regardless, it seems good to be conservative here.

Raising the limit in the future is a nonbreaking change, but lowering it can break peoples' builds!